pkg/framework/plugins/nodeutilization/nodeutilization.go

```diff
@@ -238,10 +219,11 @@ func evictPodsFromSourceNodes(
 	podFilter func(pod *v1.Pod) bool,
 	resourceNames []v1.ResourceName,
 	continueEviction continueEvictionCond,
+	usageSnapshot usageClient,
 ) {
 	// upper bound on total number of pods/cpu/memory and optional extended resources to be moved
 	totalAvailableUsage := map[v1.ResourceName]*resource.Quantity{}
-	for _, resourceName := range resourceNames {
+	for _, resourceName := range usageSnapshot.resourceNames() {
```

Comment:
Wouldn't it be better to use only one source of truth for the resource names. Either from the client or from the plugin struct?

Reply:
Is it useful to get the resourceNames from the client/snapshot? Can it change over time?

Reply:
I see. I think it would be better not channel it through the usageSnapshot/usageClient and use directly. We can still pass it to the client but don't have to expose it through it.